app/models/system_notification.rb

```diff
@@ -50,6 +51,6 @@ def self.last_sync_timestamp
     SystemNotification
       .order(last_time_changed_at: :desc)
       .first&.last_time_changed_at&.to_i ||
-      User.order(created_at: :desc).first&.created_at&.to_i
+      User.order(created_at: :asc).limit(1).first&.created_at&.to_i
```

Comment:
doesn't '.first' use LIMIT inside?

Reply:
Ahh, sure it does. Thank you!

Reply:
True. For array with 100k users `ObjectSpace.memsize_of(obj)` returns `780 KB`. And limit for items nr in an array should be around 500mio. 
Do you see any issues?